packages/beta/src/__tests__/api/filesMultiPart.int.spec.ts

```diff
@@ -0,0 +1,296 @@
+    const response = await client.filesMultiPart.multipartUploadSession(
+      {
+        ...getFileCreateArgs({ name: testfile }).localFileMeta,
+      },
+      numberOfParts,
+      false
+    );
+
+    await expect(response.uploadPart(0, fileChunks[0])).resolves.not.toThrow();
+    await expect(response.uploadPart(0, fileChunks[0])).resolves.not.toThrow();
+  });
+  const retryCases = [
+    { numberOfParts: 1, partsToFail: 0 },
+    { numberOfParts: 5, partsToFail: 3 },
+    { numberOfParts: 10, partsToFail: 5 },
+  ];
+  it.each(retryCases)(
+    'can retry failed parts',
+    async ({ numberOfParts, partsToFail }) => {
+      const fileChunks = divideFileIntoChunks(testfile, numberOfParts);
+      const response = await client.filesMultiPart.multipartUploadSession(
+        {
+          ...getFileCreateArgs({ name: testfile }).localFileMeta,
+        },
+        numberOfParts,
+        false
+      );
+      const allParts = Array.from({ length: numberOfParts }, (_, i) => i);
+      expect(response.getNotCompletedParts()).toEqual(allParts);
+
+      for (let i = 0; i < numberOfParts; i++) {
+        console.log(`Uploading part ${i + 1}`);
+        if (i === partsToFail) {
+          // Fail  manually
+          try {
+            await response.uploadPart(i, null);
+          } catch (error) {
+            console.log(`part ${i + 1} upload failed ${error}`);
+          }
+        } else {
+          await response.uploadPart(i, toArrayBuffer(fileChunks[i]));
+        }
+      }
+
+      expect(response.getNotCompletedParts()).toEqual([partsToFail]);
+
+      await response.uploadPart(partsToFail, fileChunks[partsToFail]);
+      expect(response.getNotCompletedParts()).toEqual([]);
+      console.log(
+        `Uploaded file  ${response.multiPartFileUploadResponse.id} successfully.`
+      );
+      await sleepPromise(2000);
+      const [retrievedFile] = await client.files.retrieve([
+        { id: response.multiPartFileUploadResponse.id },
+      ]);
+      expect(retrievedFile.uploaded).toBeTruthy();
+    }
+  );
+  test('can concurrently upload parts', async () => {
+    const numberOfParts = 10;
+    const fileChunks = divideFileIntoChunks(testfile, numberOfParts);
+    const response = await client.filesMultiPart.multipartUploadSession(
+      {
+        ...getFileCreateArgs({ name: testfile }).localFileMeta,
+      },
+      numberOfParts,
+      false
+    );
+    const allParts = Array.from({ length: numberOfParts }, (_, i) => i);
+    expect(response.getNotCompletedParts()).toEqual(allParts);
+
+    await Promise.all(
+      fileChunks.map(
+        async (fileChunk, i) => await response.uploadPart(i, fileChunk)
+      )
+    );
+
+    expect(response.getNotCompletedParts()).toEqual([]);
+    console.log(
+      `Uploaded file  ${response.multiPartFileUploadResponse.id} successfully.`
+    );
+    await sleepPromise(2000);
+    const [retrievedFile] = await client.files.retrieve([
```

Comment:
Repeat also here, to handle cases where upload=true takes a long time.

Reply:
done above